lib/ext/core/oauth/authorize.ex

```diff
@@ -0,0 +1,24 @@
+defmodule Ext.Oauth.Authorize do
+  require IEx
+
+  def call(
+        %{payload: payload, provider: provider} = args,
+        repo,
+        %{user: user_schema, auth: auth_schema} = schemas,
+        form
+      ) do
+    uid = Ext.Auth.GetUserUniqKey.call(provider, payload)
+
+    user_assoc = Ext.Ecto.Schema.get_schema_assoc(auth_schema, user_schema)
+
+    case auth_schema
+         |> repo.get_by(%{provider: provider, uid: uid})
+         |> repo.preload(user_assoc) do
+      nil ->
+        Ext.Oauth.SignUp.call(args, provider, uid, repo, schemas, form)
```

Comment:
repo, schemas, form - might be one map. Because we later will add 2/3 new arguments